### PR TITLE
add mathematical boundary types : HOEXTRAPLI

### PR DESCRIPTION
**Pull request by DUGKS**
add a new BC type to fill ghost CELL value. [#579](https://github.com/AMReX-Codes/amrex/issues/579)

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
Thanks for the pull-request!  Could you add something like `TOCC` or `CC` in the BC type name so that it is more explicit that the value is extrapolated to cell center?

**Comment by DUGKS** [comment]
> Thanks for the pull-request! Could you add something like `TOCC` or `CC` in the BC type name so that it is more explicit that the value is extrapolated to cell center?

Done. 
HOEXTRAPCCLI is too wordy so I just use HOEXTRAPCC. But it may raise ambiguity since users would expect this extrapolation scheme the same as the one used for EDGE values.
